lib/components/RangePicker.vue

```diff
@@ -0,0 +1,326 @@
+<script lang="ts">
+import { defineComponent, VNode, DirectiveBinding, PropType } from 'vue'
+import { faGripLinesVertical } from '@fortawesome/free-solid-svg-icons/faGripLinesVertical'
+import { clamp, get, has, invoke, round } from 'lodash'
+
+import Fa, { library } from './Fa'
+
+import type { Variant } from '@/types'
+
+/**
+ * A component to wrap an HTML element with a range picker overlay.
+ */
+export default defineComponent({
+  name: 'RangePicker',
+  components: {
+    Fa
+  },
+  directives: {
+    draggable: {
+      inserted(el: HTMLElement, binding: DirectiveBinding, vnode: VNode): void {
+        let startX: number, initialMouseX: number
+        const relative = binding.modifiers?.relative ?? false
+
+        // Emit an event to the parent component
+        function emitEvent({ name, data }: { name: string; data: any }) {
+          const handlers = get(vnode, 'data.on') ?? get(vnode, 'componentOptions.listeners')
+
+          if (has(handlers, name)) {
+            invoke(handlers, `${name}.fns`, data)
+          }
+        }
+
+        // Handle the dragging of the element
+        function mousemove(event: MouseEvent) {
+          const clientX = event.clientX
+          const offset = relative ? el.offsetWidth : 0
+          const maxX = (vnode.context as any).rangeWidth() - offset
+          const data = clamp(startX + clientX - initialMouseX, 0, maxX)
+          emitEvent({ name: 'dragged', data })
+          return false
+        }
+
+        // Clean up listeners once the dragging ends
+        function mouseup() {
+          document.removeEventListener('mousemove', mousemove)
+          document.removeEventListener('mouseup', mouseup)
+        }
+
+        // Register the drag event handlers
+        el.addEventListener('mousedown', (event: MouseEvent) => {
+          startX = el.offsetLeft
+          initialMouseX = event.clientX
+          document.addEventListener('mousemove', mousemove)
+          document.addEventListener('mouseup', mouseup)
+          return false
+        })
+      }
+    }
+  },
+  model: {
+    prop: 'value',
+    event: 'update'
+  },
+  props: {
+    /**
+     * Represents the initial values of the range bounds. Should contain two numbers
```

Comment:
In other component props doc we have more direct descriptions without the verbs.
```suggestion
     * Initial values of the range bounds. Should contain two numbers.
```